spec/support/query_counter.rb

```diff
@@ -5,10 +5,11 @@ def self.count(&block)
     new.count(&block)
   end
 
-  IGNORED_STATEMENTS = %w(CACHE SCHEMA)
+  IGNORED_STATEMENTS = %w(CACHE SCHEMA).freeze
+  OTHER_SQL = %w(ROLLBACK BEGIN COMMIT).freeze
 
   def callback(_name, _start, _finish, _id, payload)
-    @count += 1 unless IGNORED_STATEMENTS.include?(payload[:name])
+    @count += 1 unless IGNORED_STATEMENTS.include?(payload[:name]) || payload[:name].nil? && OTHER_SQL.include?(payload[:sql])
```

Comment:
I tend to shy away from `unless`, but the if was very confusing for me

  name|sql|it counts?
  ----|---|------
  CACHE|*|no
  SCHEMA|*|no
  nil|ROLLBACK|no
  nil|BEGIN|no
  nil|COMMIT|no
  nil|*|yes
  *|*|yes

I'm coming up with the following:

```ruby
# option 1
if !IGNORED_STATEMENTS.include?(payload[:name]) &&
   (payload[:name].present? || !OTHER_SQL.include?(payload[:sql]))
  @count += 1
end

# option 2
unless IGNORED_STATEMENTS.include?(payload[:name]) ||
       (payload[:name].nil? && OTHER_SQL.include?(payload[:sql]))
  @count += 1
end
```